internal/alloydb/refresh.go

```diff
@@ -113,99 +114,82 @@ func fetchEphemeralCert(
 	}
 	resp, err := cl.GenerateClientCertificate(ctx, req)
 	if err != nil {
-		return certChain{}, errtype.NewRefreshError(
+		return nil, errtype.NewRefreshError(
 			"create ephemeral cert failed",
 			inst.String(),
 			err,
 		)
 	}
-	// There should always be two certs in the chain. If this fails, the API has
-	// broken its contract with the client.
-	if len(resp.PemCertificateChain) != 2 {
-		return certChain{}, errtype.NewRefreshError(
-			"missing instance and root certificates",
+
+	certChainPEM := append([]string{resp.PemCertificate}, resp.PemCertificateChain...)
+	certPEMBlock := []byte(strings.Join(certChainPEM, "\n"))
+	keyPEMBlock := &pem.Block{
+		Type: "RSA PRIVATE KEY",
+		Bytes: x509.MarshalPKCS1PrivateKey(key),
+	}
+
+	cert, err := tls.X509KeyPair(certPEMBlock, pem.EncodeToMemory(keyPEMBlock))
+	if err != nil {
+		return nil, errtype.NewRefreshError(
+			"create ephemeral cert failed",
 			inst.String(),
 			err,
 		)
 	}
-	rc, err := parseCert(resp.PemCertificateChain[1]) // root cert
-	if err != nil {
-		return certChain{}, errtype.NewRefreshError(
-			"failed to parse root cert",
+
+	// TODO(fixme) Take the root cert from the cert chain for now.
+	caCertPEMBlock, _ := pem.Decode([]byte(certChainPEM[2]))
+	if caCertPEMBlock == nil {
+		return nil, errtype.NewRefreshError(
+			"create ephemeral cert failed",
 			inst.String(),
-			err,
+			errors.New("no PEM data found in the ca cert"),
 		)
 	}
-	ic, err := parseCert(resp.PemCertificateChain[0]) // intermediate cert
+	caCert, err := x509.ParseCertificate(caCertPEMBlock.Bytes)
 	if err != nil {
-		return certChain{}, errtype.NewRefreshError(
-			"failed to parse intermediate cert",
+		return nil, errtype.NewRefreshError(
+			"create ephemeral cert failed",
 			inst.String(),
 			err,
 		)
 	}
-	c, err := parseCert(resp.PemCertificate) // client cert
+
+	// Extract expiry
+	clientCertPEMBlock, _ := pem.Decode([]byte(certChainPEM[0]))
+	if clientCertPEMBlock == nil {
+		return nil, errtype.NewRefreshError(
+			"create ephemeral cert failed",
+			inst.String(),
+			errors.New("no PEM data found in the client cert"),
+		)
+	}
+	clientCert, err := x509.ParseCertificate(clientCertPEMBlock.Bytes)
 	if err != nil {
-		return certChain{}, errtype.NewRefreshError(
-			"failed to parse client cert",
+		return nil, errtype.NewRefreshError(
+			"create ephemeral cert failed",
 			inst.String(),
 			err,
 		)
 	}
 
-	return certChain{
-		root:         rc,
-		intermediate: ic,
-		client:       c,
+	return &certs{
+		certChain: &cert,
+		caCert: caCert,
+		expiry: clientCert.NotAfter,
 	}, nil
 }
 
 // createTLSConfig returns a *tls.Config for connecting securely to the AlloyDB
 // instance.
-func createTLSConfig(inst InstanceURI, cc certChain, info connectInfo, k *rsa.PrivateKey) *tls.Config {
-	certs := x509.NewCertPool()
-	certs.AddCert(cc.root)
+func createTLSConfig(inst InstanceURI, cc *certs, info connectInfo) *tls.Config {
```

Comment:
Let's just inline this function now that it's tiny.

Reply:
Okay

Reply:
Done